src/app/components/match-list/match-list.component.html

```diff
@@ -1,3 +1,28 @@
+<table mat-table [dataSource]="setupDataSource()"  multiTemplateDataRows class="mat-elevation-z8">
```

Comment:
I know that this way of setting up the datasource isn't pretty, but it was the only way I could get it to work. Do you have a suggestion of how I can do this in a better way? @nilssonemil

Reply:
I think the html should refer to a member of the component, which would be initialized in ngInit which is called before the first render. I suppose this is some kind of workaround to update the data source? In this case I would probably look at https://angular.io/guide/component-interaction to find a way to communicate the data change between the this component and the one owning the data.

Reply:
Yes it is a very bad workaround that I have now realized is never gonna work. Thank you for the link, hopefully I can find what I need there.